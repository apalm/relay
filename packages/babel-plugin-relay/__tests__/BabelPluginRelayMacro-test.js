/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+relay
 */

'use strict';

require('configureForRelayOSS');

const babel = require('babel-core');

describe('BabelPluginRelayMacro', () => {
  test('works', () => {
    const basic = `
      'use strict';

      const {graphql} = require('../../react-relay/modern/ReactRelayGraphQL.macro');
      const CompatProfilePic = require('CompatProfilePic');

      const CompatViewerQuery = graphql\`
        query CompatViewerQuery($id: ID!, $scale: Float = 1.5) {
          node(id: $id) {
            ... on User {
              id
              ...CompatProfilePic_user
            }
          }
        }
      \`;
    `;
    const {code} = babel.transform(basic, {
      plugins: ['babel-macros'],
      filename: __filename,
      compact: false,
      parserOpts: {plugins: ['jsx']},
      babelrc: false,
    });
    expect(code).toMatchSnapshot();
  });
});
