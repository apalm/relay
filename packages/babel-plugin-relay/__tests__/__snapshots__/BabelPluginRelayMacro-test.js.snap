// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BabelPluginRelayMacro works 1`] = `
"
'use strict';

const CompatProfilePic = require('CompatProfilePic');

const CompatViewerQuery = function () {
  return require('./__generated__/CompatViewerQuery.graphql');
};"
`;
